ostelco-ios-client/Coordinators/RootCoordinator.swift

```diff
@@ -35,21 +53,81 @@ class RootCoordinator {
         self.window.rootViewController = newRoot
     }
     
-    func showLogin() {
-        let loginViewController = LoginViewController.fromStoryboard()
-        self.topViewController?.present(loginViewController, animated: true)
+    func determineAndNavigateToInitialDestination(animated: Bool = true) {
+        self.determineDestination()
+            .done { [weak self] destination in
+                self?.navigate(to: destination, from: nil, animated: animated)
+            }
+            .catch { error in
+                ApplicationErrors.log(error)
+                // TODO: Figure out how to deal with this failing.
+            }
     }
     
-    func showEmailEntry() {
-        guard let emailNav = Storyboard.email.asUIStoryboard.instantiateInitialViewController() else {
-            ApplicationErrors.assertAndLog("Could not instantiate email nav!")
-            return
+    func determineDestination() -> Promise<RootCoordinator.Destination> {
+        guard self.userManager.hasCurrentUser else {
+            // NOPE! We need to log in.
+            return .value(.login)
+        }
+        
+        guard UserDefaultsWrapper.pendingEmail == nil else {
+            // We still need to confirm the user's email
+            return .value(.email)
         }
         
-        self.topViewController?.present(emailNav, animated: true)
+        return APIManager.shared.primeAPI
+            .loadContext()
+            .map { context -> RootCoordinator.Destination in
+                self.userManager.customer = context.customer
+                guard let region = context.getRegion() else {
+                    return .country
+                }
+                
+                OnBoardingManager.sharedInstance.region = region
+                return self.handleRegionResponse(region)
+            }
+            // Recover allows us to check for an error but continue the chain
+            .recover { error -> Promise<RootCoordinator.Destination> in
+                switch error {
+                case APIHelper.Error.jsonError(let requestError):
+                    if requestError.httpStatusCode == 404 {
+                        return .value(.signUp)
+                    } // else, keep going.
+                default:
```

Comment:
There was another case here before but we don't need it because the API we're using here returns the `jsonError` type